raiden/api/python.py

```diff
@@ -637,6 +637,44 @@ def get_network_events(
             to_block=to_block,
         ), key=lambda evt: evt.get('block_number'), reverse=True)
 
+    def get_channel_history_events(
+            self,
+            token_address: typing.TokenAddress,
+            partner_address: typing.Address = None,
+            from_block: typing.BlockSpecification = 0,
+            to_block: typing.BlockSpecification = 'latest',
+    ):
+        if not is_binary_address(token_address):
+            raise InvalidAddress(
+                'Expected binary address format for token in get_token_network_events',
+            )
+
+        returned_events = []
+        raiden_events = self.raiden.wal.storage.get_events_by_block(
+            from_block=from_block,
+            to_block=to_block,
+        )
+
+        # Here choose which raiden internal events we want to expose to the end user
+        for block_number, event in raiden_events:
+            if isinstance(event, EVENTS_EXTERNALLY_VISIBLE):
+                new_event = {
+                    'block_number': block_number,
+                    'event': type(event).__name__,
+                }
+                new_event.update(event.__dict__)
+                if not partner_address:
+                    returned_events.append(new_event)
```

Comment:
That way you still do the work before you know if it's necessary. Please filter first, and then update the event.